noxfile.py

```diff
@@ -147,26 +151,33 @@ def _get_dependencies_from_pyproject_toml(extras: str | None = None):
 @nox.session
 def requirements(session: nox.Session) -> None:
     """
-    Regenerate the pinned requirements for running tests and building
+    Regenerate the locked requirements for running tests and building
     documentation.
 
     This workflow updates :file:`uv.lock` to contain pinned requirements
     for different versions of Python, different operating systems, and
-    different dependency sets (i.e., `docs` or `tests`).
+    different dependency sets.
 
-    When run in CI, this session will create a file that contains the
-    pull request message for the GitHub workflow that updates the pinned
+    When run in CI, this session creates a file that contains the pull
+    request message for the GitHub workflow that updates the pinned
     requirements (:file:`.github/workflows/update-pinned-reqs.yml`).
     """
-    uv_lock_upgrade = ["uv", "lock", "--upgrade", "--no-progress"]
 
-    # When silent is `True`, `session.run()` returns a multi-line string
-    # with the standard output and standard error.
+    # Running `uv lock` fails when `uv.lock` has a merge conflict or
+    # is otherwise invalid. To avoid situations like this, preemptively
+    # delete `uv.lock`. Since `uv lock` uses information in `uv.lock`
+    # even if `--upgrade` is specified, deleting `uv.lock` may make
+    # running `uv lock` more deterministic.
+
+    lockfile = pathlib.Path(root_dir / "uv.lock")
+    lockfile.unlink(missing_ok=True)
```

Comment:
I considered deleting `uv.lock` only when running `uv lock` locally or when a git merge is underway, but decided it's generally better to do the same thing locally as in CI most of the time.

Reply:
Deleting `uv.lock` removes pins and other settings that are not included in `pyproject.toml`, but we want `pyproject.toml` to be the definitive source for requirements information, so in our case this should be fine.